pkg/apply/apply.go

```diff
@@ -988,6 +1040,8 @@ func (t *TopicApplier) updatePlacementRunner(
 			}
 			return err
 		}
+		// add updated replica assignments to changes tracker
+		changes.mergeReplicaAssignments(assignmentsToUpdate[i:end])
```

Comment:
I think it would be a smart idea, to keep this code easier to udnerstand and digest, to move the place where you record the changes into the ChangeTracker right next to the code that prints the same information on the logs. 
That would make it trivial to spot whether the two happen in the same place and if one has been forgotten.

It also would make it simpler to refactor the change tracker into a `Output` interface where the event is recorded. This implementation then takes care of logging and to track the update for json. It would guarantee the log and json are in sync without having to verify the code scattered around.